src/roman_datamodels/datamodels/_core.py

```diff
@@ -305,24 +295,30 @@ def get_crds_parameters(self):
         -------
         dict
         """
-        crds_header = {
+        return {
             key: val
             for key, val in self.to_flat_dict(include_arrays=False).items()
             if isinstance(val, (str, int, float, complex, bool))
         }
-        return crds_header
 
     def validate(self):
         """
         Re-validate the model instance against the tags
         """
         validate.value_change(self._instance, pass_invalid_values=False, strict_validation=True)
 
+    @property
+    def asdf(self):
```

Comment:
Making this public might give the wrong impression that the underlying asdf file can be accessed and modified as part of the `DataModel` interface. This does not appear to be the case, assuming we have a model `m`
```
m.asdf['my_info'] = 42
assert m.asdf['my_info'] == 42
m.save('foo.asdf')
with asdf.open('foo.asdf') as af:
    af['my_info']  # KeyError as 'my_info' was not saved
```
This property (or at least the handling of `_asdf is None` appears needed for `info` `search` and `schema_info` below. Are there options for keeping this private?

Reply:
This has now been addressed using a decorator.